src/components/AllReviews.js

```diff
@@ -1,31 +1,46 @@
 import { useEffect, useState } from 'react';
 import { getReviews } from '../utils/api';
 import { Link } from 'react-router-dom';
+import Categories from './Categories';
 
 function AllReviews() {
   const [reviews, setReviews] = useState([]);
+  const [category, setCategory] = useState('');
+
   useEffect(() => {
     getReviews().then((reviewsFromAPI) => {
-      setReviews(reviewsFromAPI);
+      if (category === '') {
+        setReviews(reviewsFromAPI);
+      } else {
+        let reviewsByCategory = reviewsFromAPI.filter(
+          (review) => review.category === category
```

Comment:
You've built an API capable of taking in queries which we should make use of here, SQL is far more efficient than JS when it comes to sorting the data we need.
You should alter your getReviews component so it can take in an optional category argument which can be appended to your API request as a query